text/0000-opencensus-migration-plan.md

```diff
@@ -0,0 +1,141 @@
+
+### Requirements for Metrics and Trace bridges
+
+Bridges:
+
+* MUST NOT require OpenCensus to depend on OpenTelemetry
+* MUST require few or no changes to OpenCensus
+* MUST convert OpenCensus semantic conventions to OpenTelemetry semantic conventions for metric and span names and attributes
+* are NOT REQUIRED to support the entire OpenCensus instrumentation API Surface.
+
+Trace Bridges:
+
+* MUST support context propagation between OpenCensus and OpenTelemetry instrumentation
+* MUST preseve span contents where an OpenTelemetry equivalent exists
+
+Metric Bridges (to be proposed separately):
+
+* MUST be compatible with push and pull exporters
+* SHOULD support Gauges, Counters, Cumulative Histograms, and Summaries
+* are NOT REQUIRED to support Gauge Histograms
+* SHOULD support exemplars
+
+### Migrating an Application using Bridges
+
+Starting with an application using entirely OpenCensus instrumention for traces and metrics, it can be migrated by:
+
+1. Migrate the exporters (SDK)
+    1. Install the OpenTelemetry SDK, with an equivalent exporter
+    2. Install equivalent OpenTelemetry resource detectors
+    3. Install OpenTelemetry propagators for OpenCensus' `TextFormat` and `BinaryPropogator` formats.
+    4. **breaking**: Install the metrics and trace bridges
+    5. Remove initialization of OpenCensus exporters
+2. Migrate the instrumentation (API)
+    1. **breaking**: For OpenCensus instrumentation packages, migrate to the OpenTelemetry equivalent.
+    2. For external dependencies, wait for it to migrate to OpenTelemetry, and update the dependency.
+    3. For instrumentation which is part of the application, migrate it following the guidance below.
+3. Clean up: Remove the metrics and trace bridges
+
+### Migrating Libraries using OC Instrumentation
+
+#### In-place Migration
+
+Libraries which want a simple migration can choose to replace instrumentation in-place.
+
+Starting with a library using OpenCensus Instrumentation:
+
+1. Annouce to users the library's transition from OpenCensus to OpenTelemetry, and recommend users adopt OC bridges.
+2. Change unit tests to use the OC bridges, and use OpenTelemetry unit testing frameworks.
+3. After a notification period, migrate instrumentation line-by-line to OpenTelemetry. The notification period should be long for popular libraries.
+4. Remove the OC bridge from unit tests.
+
+#### Migration via Config
+
+Libraries which are eager to add native OpenTelemetry instrumentation sooner, and/or want to
+provide extended support for OpenCensus may choose to provide users the option to use OpenCensus
+instrumentation _or_ OpenTelemetry instrumentation.
+
+Starting with a library using OpenCensus Instrumentation:
+
+1. Add configuration allowing users to enable OpenTelemetry instrumentation and disable OpenCensus instrumentation.
+2. Add OpenTelemetry instrumentation gated by the configuration.
```

Comment:
Should we document the risk of discrepancy/drift between the OC/OTel instrumentation in this approach?

Reply:
Good point.  I've updated this to suggest that the two should share unit tests, as recommended in the "in-place" migration.  That should resolve this issue.